### ISSUE TITLE
Error in dims - 1L : non-numeric argument to binary operator: omp()

### ISSUE THREAD

**Issue by hansk0812** [getspams/spams-R#3]
I'm trying to run the example from the documentation: https://thoth.inrialpes.fr/people/mairal/spams/doc-R/html/index.html and I'm running into the following error. 
```r
X = matrix(rnorm(64 * 100000),nrow = 64,ncol = 100000,byrow = FALSE)
D = matrix(rnorm(64 * 200),nrow = 64,ncol = 200,byrow = FALSE)
D = D / matrix(rep(sqrt(colSums(D*D)),nrow(D)),nrow(D),ncol(D),byrow=T)
L = 10
eps =0.1
numThreads = -1

tic = proc.time()
alpha = spams.omp(X,D,L=L,eps=eps,return_reg_path = FALSE,numThreads = numThreads)
tac = proc.time()
t = (tac - tic)[['elapsed']]
print("%f signals processed per second\n",as.double(ncol(X)) / t)

########################################
# Regularization path of a single signal
########################################
X = matrix(rnorm(64 * 1),nrow = 64,ncol = 1,byrow = FALSE)
D = matrix(rnorm(64 * 10),nrow = 64,ncol = 10,byrow = FALSE)
D = D / matrix(rep(sqrt(colSums(D*D)),nrow(D)),nrow(D),ncol(D),byrow=T)
L = 5
res = spams.omp(X,D,L = L,eps = eps,return_reg_path = TRUE,numThreads = numThreads)
```
```r
Error in dims - 1L : non-numeric argument to binary operator
``` 
I've checked the input types like so:

```r
> class(X); class(D); class(L); class(eps); class(numThreads)
[1] "matrix"
[1] "matrix"
[1] "numeric"
[1] "numeric"
[1] "numeric"
```

**Comment by gdurif** [comment]
Hi @hansk0812 , thanks for the report, we will investigate the issue and get back to you ASAP.
Best

**Comment by gdurif** [comment]
Fixed by #6
